stglib/sg/cdf2nc.py

```diff
@@ -0,0 +1,208 @@
+    ds = utils.create_water_level_var(ds)
+    ds = utils.create_water_depth_var(ds)
+    ds = utils.ds_add_lat_lon(ds)
+    ds = utils.add_start_stop_time(ds)
+    ds = utils.add_delta_t(ds)
+    ds = utils.add_min_max(ds)
+
+    # Write to .nc file
+    print("Writing cleaned/trimmed data to .nc file")
+    nc_filename = ds.attrs["filename"] + "s-tide-a.nc"
+
+    ds.to_netcdf(
+        nc_filename, unlimited_dims=["time"], encoding={"time": {"dtype": "i4"}}
+    )
+    utils.check_compliance(nc_filename, conventions=ds.attrs["Conventions"])
+
+    print(f"Done writing netCDF file {nc_filename}")
+
+
+def ds_rename_vars(ds):
+    """
+    Rename variables to be CF compliant
+    """
+    varnames = {"Temp": "T_28"}
+
+    # Check to make sure they exist before trying to rename
+    newvars = {}
+    for k in varnames:
+        if k in ds:
+            newvars[k] = varnames[k]
+    return ds.rename(newvars)
+
+
+def ds_drop_tid(ds):
+    """
+    Drop global attribute metadata not needed for .tid file
+    """
+    gatts = [
+        "WaveInterval",
+        "WaveIntervalUnits",
+        "WaveSamples",
+        "sample_rate",
+        "sample_rate_units",
+        "BurstDuration",
+        "BurstDurationUnits",
+        "WaveBurstsPerDay",
+        "NumberOfWaveBursts",
+    ]
+
+    # Check to make sure they exist
+    for k in gatts:
+        if k in ds.attrs:
+            del ds.attrs[k]
+    return ds
+
+
+def ds_drop_wb(ds):
+    """
+    Drop global attribute metadata not needed for .wb file
+    """
+    gatts = [
+        "WaveInterval",
+        "WaveIntervalUnits",
+        "WaveSamples",
+        "sample_rate",
+        "sample_rate_units",
+        "BurstDuration",
+        "BurstDurationUnits",
+        "WaveBurstsPerDay",
+        "NumberOfWaveBursts",
+        "TideInterval",
+        "TideIntervalUnits",
+        "TideDuration",
+        "TideDurationUnits",
+        "TideSamplesPerDay",
+        "NumberOfTideMeasurements",
+    ]
+
+    # Check to make sure they exist
+    for k in gatts:
+        if k in ds.attrs:
+            del ds.attrs[k]
+    return ds
+
+
+def avg_tide_bursts(ds):
+
+    # Calculate how many rows to subdivide each wave burst (round up)
+    rows = math.ceil(
+        float(ds.attrs["BurstDuration"]) / float(ds.attrs["calculated_tide_interval"])
+    )
+
+    # Calculate how many columns to subdivide each wave burst
+    cols = int(
+        float(ds.attrs["calculated_tide_interval"]) * float(ds.attrs["sample_rate"])
+    )
+
+    # Calculate number of values to average based on tide duration
+    values_avg = int(
+        float(ds.attrs["calculated_tide_duration"]) * float(ds.attrs["sample_rate"])
+    )
+
+    # Define time interval
+    delta_t = int(ds.attrs["calculated_tide_interval"])
+    delta_t = str(delta_t) + "s"
+
+    # Reshape P_1 and P_1ac pressure bursts and average
+    P1_avg = []
+    P1ac_avg = []
+    timestamp = []
+    for t in range(0, ds["time"].size):
```

Comment:
This type of looping is often not necessary with coordinates associated with data in an xarray Dataset. Is there a variable number of nans at the end of the array throughout the data?

Reply:
Yes, some rows need to be filled with nans, but others don't.